pybossa/api/__init__.py

```diff
@@ -417,6 +420,82 @@ def task_progress(project_id=None, short_name=None):
     return Response(json.dumps(task_count_dict), mimetype="application/json")
 
 
+@jsonpify
+@login_required
+@blueprint.route('/preferences/<user_name>', methods=['GET'])
```

Comment:
Any reason not to reuse the Pybossa API class - https://github.com/bloomberg/pybossa/blob/master/pybossa/api/user.py? Our task, project, user and result API are updated through their API class, should we keep consistency on the code?

Reply:
The code was following convention in the existing file. If this should be changed, could you describe or show an example? Thank you.

Reply:
User API naturally has GET and PUT method. 
Get: `curl ../api/user/id` will return all information of the user, including user profile.
Put: `curl ../api/user/id -d data` can update user information. However, it fails to update user profile because this field is a nested json payload

We can consider `_update_instance` to allow User API to update user profile https://github.com/bloomberg/pybossa/blob/e3114b3f719575aec9920cacfba3ebc9e619ca5e/pybossa/api/api_base.py#L451

Reply:
The new routes are related to user preferences, rather than the user. (`/api/preferences/username`), which lends the code to this file instead of the UserAPI. I'm not sure if using the UserApi class would be a cleaner approach; this implementation is relatively simple and following convention in the existing file.